### PR TITLE
[libSyntax][Cmake] Fix gyb invocation quoting issue on Windows

### PR DESCRIPTION
**Pull request by AndrewSB**
The lib/Syntax build system assumed a *nix quoting style for arguments
passed into gyb. That assumption breaks on Windows. This patch updates
the build system to Do The Right Thing™ when built on Windows.

### CODE REVIEW

**Review by harlanhaskins — APPROVED** [review]
Thanks!

**Comment by harlanhaskins** [comment]
@swift-ci please test

**Review by harlanhaskins — CHANGES_REQUESTED** [review]
Turns out this change makes the line directive become:

`"#line %(line)d "%(file)s""`, which no longer parses.

File: lib/Syntax/CMakeLists.txt [hunk 1]
```diff
@@ -1,6 +1,7 @@
 set(line_directive "#line" "%(line)d" "\"%(file)s\"")
+
 set(SWIFT_GYB_FLAGS
-  --line-directive "'${line_directive}'")
+  --line-directive "\"${line_directive}\"")
```
Comment (jrose-apple) [inline lib/Syntax/CMakeLists.txt]: I don't think this is correct; note that the actual body of `${line_directive}` contains quotes.

Comment (harlanhaskins) [reply lib/Syntax/CMakeLists.txt]: @AndrewSB you might need to double-escape the quotes.

`”\\\”%(file)s\\\””`

Comment (AndrewSB) [reply lib/Syntax/CMakeLists.txt]: parsing on parsing 😂

Thanks you two, I'll make the change now, and I can test it tomorrow during the day, when I'm back at my Windows machine

**Comment by AndrewSB** [comment]
Made that change, can one of you ask CI to run again? But please don't merge until I verify that this continues to work on Windows.

**Comment by harlanhaskins** [comment]
@swift-ci please smoke test

**Comment by harlanhaskins** [comment]
Nice! Okay, now I wonder if CMake has some kind of “escape quotes” function we could use, instead of thjs

**Comment by compnerd** [comment]
I don't think that there is any special quoting function. Maybe @llvm-beanz knows?

**Comment by AndrewSB** [comment]
This currently breaks on windows, working on a fix now

**Comment by AndrewSB** [comment]
This worked on the latest clean build I tried, @harlanhaskins: can you re-review and run CI please?

File: cmake/modules/SwiftHandleGybSources.cmake [hunk 2]
```diff
@@ -57,24 +59,29 @@ function(handle_gyb_source_single dependency_out_var_name)
...
-          "${GYB_SINGLE_OUTPUT}.tmp" "${GYB_SINGLE_OUTPUT}"
+        "${CMAKE_COMMAND}" -E copy_if_different "${GYB_SINGLE_OUTPUT}.tmp" "${GYB_SINGLE_OUTPUT}"
       COMMAND
-          "${CMAKE_COMMAND}" -E remove "${GYB_SINGLE_OUTPUT}.tmp"
-      OUTPUT "${GYB_SINGLE_OUTPUT}"
-      DEPENDS "${gyb_tool_source}" "${GYB_SINGLE_DEPENDS}" "${GYB_SINGLE_SOURCE}" "${gyb_expand_deps}"
-      COMMENT "Generating ${basename} from ${GYB_SINGLE_SOURCE} ${GYB_SINGLE_COMMENT}"
-      WORKING_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}"
-      SOURCES "${GYB_SINGLE_SOURCE}"
-      IDEMPOTENT)
```
Comment (harlanhaskins) [inline cmake/modules/SwiftHandleGybSources.cmake]: I'd really rather not reformat this file in this patch, just to make it easier to verify subtle differences didn't alter the CMake semantics.

**Comment by harlanhaskins** [comment]
Rest of the patch looks good 👍.

@swift-ci please clean test

File: cmake/modules/SwiftHandleGybSources.cmake (hunk 2 above)
Comment (AndrewSB) [reply cmake/modules/SwiftHandleGybSources.cmake]: I'll put it into a separate commit. Do you want me to make a new PR with it, or is keeping it in this one OK?

Comment (harlanhaskins) [reply cmake/modules/SwiftHandleGybSources.cmake]: It might(?) have broken the build as it is now? I think a new PR would be best, if you can.

Comment (AndrewSB) [reply cmake/modules/SwiftHandleGybSources.cmake]: I'll do it in a new one 😄 
It looks like the CI failed with an intermittent error, but I'm not 100% sure. Once I remove the formatting lets try again?

**Comment by AndrewSB** [comment]
@harlanhaskins moved the formatting changes into a different commit. I'll submit a PR after this is merged in so I don't have to deal with rebasing

Could you re-test?

**Comment by harlanhaskins** [comment]
@swift-ci please clean test

**Comment by harlanhaskins** [comment]
Ignore that spurious failure -- it's re-running the build.

**Comment by AndrewSB** [comment]
Another infrastructure failure? It reported seconds after you asked and I don't see an obvious gyb issue in the logs

**Comment by harlanhaskins** [comment]
@swift-ci please clean test

**Pull request closed without merging**